src/vs/server/node/server.cli.ts

```diff
@@ -345,7 +345,7 @@ export async function main(desc: ProductDescription, args: string[]): Promise<vo
 		});
 
 		if (waitMarkerFilePath) {
-			waitForFileDeleted(waitMarkerFilePath);
+			await waitForFileDeleted(waitMarkerFilePath);
```

Comment:
Is this causing a deadlock with the promise below? can you test this?

Reply:
I tested launching with `--wait` and reading from stdin.

Why do you think tis can cause a deadlock? The second promise (reading stdin) is already running. It's just awaited

Reply:
Ok good, if you tested it.